homeassistant/components/roku.py

```diff
@@ -0,0 +1,119 @@
+"""
+Support for Roku platform.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/roku/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.discovery import SERVICE_ROKU
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-roku==3.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'roku'
+
+SERVICE_SCAN = 'roku_scan'
+
+ATTR_ROKU = 'roku'
+
+DATA_ROKU = 'data_roku'
+
+NOTIFICATION_ID = 'roku_notification'
+NOTIFICATION_TITLE = 'Roku Setup'
+NOTIFICATION_SCAN_ID = 'roku_scan_notification'
+NOTIFICATION_SCAN_TITLE = 'Roku Scan'
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [vol.Schema({
+        vol.Required(CONF_HOST): cv.string
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+# Currently no attributes but it might change later
+ROKU_SCAN_SCHEMA = vol.Schema({})
+
+
+async def scan_for_rokus(hass):
+    """Scan for devices and present a notification of the ones found."""
+    from roku import Roku, RokuException
+    rokus = Roku.discover()
```

Comment:
Is this async safe, ie no non asyncio I/O being done, or blocking?

Reply:
I looked in the library, and answering my own question, no it's not async safe. A (sync) socket is used.

Either make this whole function a normal function, ie not coroutine, or schedule this call on the executor thread pool via `hass.async_add_executor_job`.